contrib/repairStatus.py

```diff
@@ -0,0 +1,199 @@
+                                "_cache": True
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+    }
+  }
+
+  return json.dumps(query_json)
+
+
+# Execute an ES _count to return the number of ranges that have finished repairing
+def calculate_range_repairs(host, port, node, start_time, current_time):
+  url = 'http://%(host)s:%(port)s/_count' % dict(host=host, port=port)
+  query = generate_range_repair_query(node, start_time, current_time)
+  
+  log.info("Ranges completed: curl %(url)s -d '%(query)s'" % dict(url=url, query=query)) 
+  r = requests.get(url, data=query)
+  
+  if r.status_code == 200:
+    response = json.loads(r.text)
+    return response['count']
+
+
+def generate_total_ranges_query(node, start_time, current_time):
+  query_json = {
+    "fields": "message",
+    "query": {
+      "filtered": {
+        "filter": {
+          "bool": {
+            "must": [
+              {
+                "range": {
+                  "@timestamp": {
+                    "to": current_time,
+                    "from": start_time
+                  }
+                }
+              },
+              {
+                "fquery": {
+                  "_cache": True,
+                  "query": {
+                    "query_string": {
+                      "query": 'host:("' + node + '")'
+                    }
+                  }
+                }
+              }
+            ]
+          }
+        },
+        "query": {
+          "query_string": {
+            "query": 'message: "Starting repair"'
+          }
+        }
+      }
+    }
+  }
+
+  return json.dumps(query_json)
+
+
+# Execute an ES _search to return the total number of ranges that need repairing
+def calculate_total_ranges(host, port, node, start_time, current_time):  
+  url = 'http://%(host)s:%(port)s/_search' % dict(host=host, port=str(port))
+  query = generate_total_ranges_query(node, start_time, current_time)
+
+  log.info("Total ranges: curl %(url)s -d '%(query)s'" % dict(url=url, query=query))
+  r = requests.get(url, data=query )
+  
+  if r.status_code == 200:
+    response = json.loads(r.text)
+    message = json.dumps(response['hits']['hits'][0]['fields']['message'])    
+    total_ranges = int(re.search('(?<=repairing\s)\d+(?=\sranges)', message).group(0))
+    return total_ranges
+
+
+def print_totals(node, days, total_successful_range_repairs, total_to_repair):
+  percent_complete = total_successful_range_repairs * 100 / total_to_repair
+  print ("Checking repair messages for", node, "for the past", days, "days")
+  print ("Count is: " + str(total_successful_range_repairs))
+  print ("Total to repair is: " + str(total_to_repair))
+  print ("Percent Complete: " + str(percent_complete) + "%")
+
+
+def main():
+  parser = argparse.ArgumentParser(
+    description = '''This script shows the current status of a repair running on a Cassandra node.  
```

Comment:
Since this is a keyword argument and not a variable being assigned, it is conventional to avoid the spaces around `=` and just have it be `description='''`.